### DISCUSSION TITLE
Project Documentation Discussion

### DISCUSSION THREAD

**Discussion by Ethan-Arrowood** [nodejs/undici#2449] [General]
What are current users, contributors, and maintainers thoughts on Undici's documentation? 

I've seen comments on a number of issues (discussions & PRs too) asking for better docs. I've received private feedback about the need for better docs. Dare I say, we can all agree we need to improve this projects documentation. 

What has proven successful for other projects? 

All ideas are good ideas, please share your thoughts 🚀

**Comment by KhafraDev** [comment]
- Moving the fetch parts of the README (spec compliance, spec divergence, etc.) into the actual documentation, so it's viewable on the docs site.
- More examples, it'd be nice to have examples that cover a majority of undici.
- [best practices](https://github.com/nodejs/undici/tree/main/docs/best-practices) for fetch. Specifically referring to the [gc](https://github.com/nodejs/undici#garbage-collection) section of the readme. Maybe also recommending undici.request when not targeting multiple environments?
- https://github.com/nodejs/undici/issues/1470 (not fetch related)

**Reply by metcoder95** [reply]
>More examples, it'd be nice to have examples that cover a majority of undici.

I would add that not only more examples are needed, but often descriptions of the concepts that `undici` handles. I've seen common issues where the problem is rooted in the _why does undici do that?_

I also find, a few bits here and there that are not documented, or just lack more context (I've been the culprit of some of those 😓).

I've always liked `jose`'s [documentation](https://github.com/panva/jose/tree/main/docs), because of how deep it goes into explaining the intrinsics of the library. Those kinds of explanations of low-level undici concepts will be extremely beneficial, as often the community is not aware of them (mostly usage of either `fetch` or `request`) until they start having issues because of not making correct usage of those concepts
